indico/MaKaC/common/search.py

```diff
@@ -30,21 +36,49 @@ def searchUsers(surName="", name="", organisation="", email="", conferenceId=Non
         # search users
         people = AvatarHolder().match(criteria, exact=exactMatch, searchInAuthenticators=searchExt)
 
-        # search authors
-        if conferenceId is not None:
-            conference = ConferenceHolder().getById(conferenceId)
-            authorIndex = conference.getAuthorIndex()
-            authors = authorIndex.match(criteria, exact=exactMatch)
-            # merge with users
-            users = people
-            people = []
-            emails = []
-            for user in users:
-                people.append(user)
-                emails.extend(user.getEmails())
-            for author in authors:
-                if author.getEmail() not in emails:
-                    people.append(author)
         return people
     else:
         return []
+
+
+def get_authors_from_author_index(event, limit):
+    """Fetch the authors of an event with the most contributions. An author is a
+    ContributionParticipation object or an Avatar object.
+
+    :param event: The event to fetch the author index from
+    :param limit: The limit of the authors to be returned
+    :return: A list of authors.
+    """
+    author_index = event.getAuthorIndex()
+    fav_users_emails = (set(itertools.chain.from_iterable(f.all_emails for f in session.user.favorite_users))
+                        if session.user else [])
+    sorted_authors = sorted(((len(values), values[0]) for author, values in author_index.iteritems()
```

Comment:
do we actually need the `len(values)` here? I think `key=len` in the `sorted()` call would do the same just fine